java/server/src/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueue.java

```diff
@@ -80,6 +88,17 @@ public boolean isReady() {
     return bus.isReady();
   }
 
+  @ManagedAttribute(name = "NewSessionQueueSize")
+  public int getQueueSize() {
+    Lock writeLock = lock.writeLock();
+    writeLock.lock();
```

Comment:
Why do we need to lock to read the queue size?

Reply:
Since it uses ConcurrentLinkedDeque, the size() requires a traversal and value may not be accurate if it is being modified when size is calculated. Hence the lock. But for exposing values if we don't need that kind of accuracy, then we can get rid of the lock. @diemol What would you suggest?